data/data-pipeline/data_pipeline/tests/score/test_score_narwhal_methods.py

```diff
@@ -0,0 +1,84 @@
+import pandas as pd
+import pytest
+from data_pipeline.config import settings
+import data_pipeline.score.field_names as field_names
+from data_pipeline.etl.score.etl_score import ScoreETL
+from data_pipeline.utils import get_module_logger
+
+logger = get_module_logger(__name__)
+
+
+@pytest.fixture
+def toy_score_df(scope="module"):
+    return pd.read_csv(
+        settings.APP_ROOT
+        / "tests"
+        / "score"
+        / "test_utils"
+        / "data"
+        / "test_drop_tracts_from_percentile.csv",
+        dtype={field_names.GEOID_TRACT_FIELD: str},
+    )
+
+
+def _helper_test_dropping_tracts(toy_score_df, drop_tracts):
+    logger.info(drop_tracts)
+    test_frame = toy_score_df[
+        ~toy_score_df[field_names.GEOID_TRACT_FIELD].isin(drop_tracts)
+    ]
+    return_df = ScoreETL._add_percentiles_to_df(
+        df=toy_score_df,
+        input_column_name="to_rank",
+        output_column_name_root="to_rank_auto",
+        drop_tracts=drop_tracts,
+    )
+
+    test_frame = test_frame.assign(
+        true_rank=test_frame["to_rank"].rank(pct=True)
+    )
+
+    check_frame = test_frame.merge(
+        return_df[
+            [
+                field_names.GEOID_TRACT_FIELD,
+                "to_rank_auto" + field_names.PERCENTILE_FIELD_SUFFIX,
+            ]
+        ],
+        on=[field_names.GEOID_TRACT_FIELD],
+    )
+
+    return check_frame["true_rank"].equals(
+        check_frame["to_rank_auto" + field_names.PERCENTILE_FIELD_SUFFIX]
+    )
+
+
+def test_drop_0_tracts(toy_score_df):
+    assert _helper_test_dropping_tracts(
+        toy_score_df, drop_tracts=[]
+    ), "Percentile in score fails when we do not drop any tracts"
```

Comment:
@mattbowen-usds you did nice lists of errors that got printed for the different configs. I can see how to emulate it here, but I'm wondering if you think I should mirror that structure?

Reply:
As above, I think these are good the way they are. Like in general, this is the better pattern to follow than what I did in those output and calculation tests --- in those, because there was so much inherent repetition and because of the place in the pipeline we were, I wanted to try to make extending the tests as painless as I could. These are much closer to true unit tests, and so having them be simple like this with a custom assertion the way you've done is better than what I did in those other examples. :1st_place_medal:

Reply:
great! thank you for the gut check!